client/lib/purchases/index.js

```diff
@@ -71,6 +73,13 @@ function getName( purchase ) {
 	return purchase.productName;
 }
 
+function getDisplayName( purchase ) {
+	if ( isJetpackProduct( purchase ) && JETPACK_PRODUCT_DISPLAY_NAMES[ purchase.productSlug ] ) {
```

Comment:
I even wonder if `isJetpackProduct` is necessary here. If a product exists in the `JETPACK_PRODUCT_DISPLAY_NAMES` list, then it's already a Jetpack product, no?